src/java.base/windows/classes/sun/nio/fs/WindowsUserDefinedFileAttributeView.java

```diff
@@ -51,8 +51,15 @@ private String join(String file, String name) {
             throw new NullPointerException("'name' is null");
         return file + ":" + name;
     }
+
     private String join(WindowsPath file, String name) throws WindowsException {
-        return join(file.getPathForWin32Calls(), name);
+        if (name == null)
+            throw new NullPointerException("'name' is null");
+        WindowsPath wp = WindowsPath.parse(file.getFileSystem(),
+                                           file + "." + name);
+        StringBuilder sb = new StringBuilder(wp.getPathForWin32Calls());
+        sb.setCharAt(sb.length() - name.length() - 1, ':');
+        return sb.toString();
```

Comment:
I suspect the right thing here is to parse "name" and maybe reject if it is not RELATIVE. Then concatenate and use createFromNormalizedPath(...) to create the WindowsPath. I expect it will be possible to reduce the number of conversions too although this isn't critical for now.

Reply:
Fixed in third commit.